app/models/package_manager/go.rb

```diff
@@ -94,8 +94,24 @@ def self.project(name)
       # that is not a valid module. https://go.dev/ref/mod#goproxy-protocol describes
       # how a $base/$module/@v/$version.mod request will return back a virtual go.mod file
       # with whatever name was passed in if there is not a go.mod file found for that version.
-      canonical_name = canonical_module_name(name)
-      search_name = canonical_name.presence || name
+      name_in_go_mod = canonical_module_name(name)
```

Comment:
Might be worth it to go ahead and rename the `canonical_module_name` method, because it's clearly both confusing and wrong to call it that, based on what we've learned now. This is just an untrusted string, that could be pure garbage, until we have checked that it matches the repo, I think?

`name_in_go_mod` as you used for the variable seems like a pretty fine name for it...

Reply:
Okies, I just didn't want touch the name in case anyone who knows more about the go mod stuff than me would object to the rename (and in case there was some meta-programming going on that was calling this method but wasn't easily greppable). Will rename unless there are complaints.